urdfModifiers/core/fixedOffsetModifier.py

```diff
@@ -0,0 +1,147 @@
+        return cls(FixedOffsetModifier.get_element_by_name(link_name, robot), robot)
+
+    @staticmethod
+    def get_element_by_name(element_name, robot):
+        """Explores the robot looking for the element whose name matches the first argument"""
+        link_list = [corresponding_link for corresponding_link in robot.links if corresponding_link.name == element_name]
+        joint_list = [corresponding_joint for corresponding_joint in robot.joints if corresponding_joint.name == element_name]
+        if len(link_list) != 0:
+            return link_list[0]
+        elif len(joint_list) != 0:
+            return joint_list[0]
+        else:
+            return None
+
+    @staticmethod
+    def get_geometry(geometry_holder):
+        """Returns the geometry type and the corresponding geometry object for a given geometry holder (visual/collision)"""
+        if geometry_holder is None:
+            return [None, None]
+        if (geometry_holder.geometry.box is not None):
+            return [geometry.Geometry.BOX, geometry_holder.geometry.box]
+        if (geometry_holder.geometry.cylinder is not None):
+            return [geometry.Geometry.CYLINDER, geometry_holder.geometry.cylinder]
+        if (geometry_holder.geometry.sphere is not None):
+            return [geometry.Geometry.SPHERE, geometry_holder.geometry.sphere]
+
+    def check_if_z_parallel(self):
+        """Validates that all Z axis of link and connected joints are parallel, otherwise raises an exception"""
+        modifier_is_valid = True
+
+        if (self.parent_joint is not None):
+            link_origin = matrix_to_xyz_rpy(self.link.visuals[0].origin)
+            modifier_is_valid = modifier_is_valid and link_origin[3] == 0 and link_origin[4] == 0
+
+        if (self.child_joint is not None):
+            child_joint_origin = matrix_to_xyz_rpy(self.child_joint.origin)
+            modifier_is_valid = modifier_is_valid and child_joint_origin[3] == 0 and child_joint_origin[4] == 0
+
+        if not modifier_is_valid:
+            raise Exception("Cannot create FixedOffsetModifier for a setup that is not z-parallel")
+
+    def get_significant_length(self):
+        """Returns the significant length, for spheres it returns diameter instead of radius"""
+        significant_length = self.link_modifier.get_significant_length()
+        geometry_type, _ = self.get_geometry(self.link_modifier.get_visual())
+        if geometry_type == Geometry.SPHERE:
+            significant_length *= 2
+        return significant_length
+
+    def get_joint_origin(self, joint):
+        """Returns the origin of a joint"""
+        return (matrix_to_xyz_rpy(joint.origin) if joint else None)
+
+    def get_link_origin(self, link):
+        """Returns the origin of a joint"""
+        return matrix_to_xyz_rpy(link.visuals[0].origin)
+
+    def calculate_offsets(self):
+        link_length = self.get_significant_length()
+        link_origin = self.get_link_origin(self.link)
```

Comment:
I find this a bit misleading. That function is not returning the link origin, but rather the origin of the frame associated to the visual element. As I understand the value of compactness in the code, do you think it could make sense to call it at least `link_visual_origin`  or something like?

Reply:
I think it makes sense for the sake of clarity